app/main.py

```diff
@@ -1,52 +1,52 @@
-import json
-import xml.etree.ElementTree as ET
-
-
-class Book:
-    def __init__(self, title: str, content: str):
-        self.title = title
-        self.content = content
-
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+from app.objects import Book
+from app.output import DisplayBook, OutputObject, PrintBook
+from app.serializers import BookJSONSerializer, BookXMLSerializer
+
+
+def display(
+        book: Book,
+        output_class: OutputObject,
+        method_type: str
+) -> None:
+    if method_type == "console":
+        output_class.output_object(book)
+    elif method_type == "reverse":
+        output_class.output_object_reverse(book)
+
+
+def print_book(
+        book: Book,
+        output_class: OutputObject,
+        method_type: str
+) -> None:
+    if method_type == "console":
+        output_class.output_object(book)
+    elif method_type == "reverse":
+        output_class.output_object_reverse(book)
+
+
+def serialize(book: Book, method_type: str) -> str:
+
+    serializer_classes = {
+        "json": BookJSONSerializer,
+        "xml": BookXMLSerializer,
+    }
+
+    serializer_class = serializer_classes[method_type]
```

Comment:
Consider adding error handling for unsupported `method_type` values to prevent potential runtime errors.